# For Image converter
camera_name: /csi_cam_0
image_topic: /image_rect_color

show_img_window: false

# Log framerate every n frames
frames_log_interval: 100

# Camera frame parameters (ground projection)
fov_width_cm: 62
fov_height_cm: 35

# Any detected objects smaller than this will not be considered
min_weed_size_cm: 2
# Any detected objects larger than this will not be considered
max_weed_size_cm: 1000

# default thresholds uses for filtering
default_weed_size_threshold_cm: 8
default_crop_size_threshold_cm: 8

############################
# Image processing configs #
############################
blur_kernel_size: 15

# Color thresholding
low_hue: 30
low_sat: 60
low_value: 50
high_hue: 90

# Morph operations
morph_size: 3
morph_opening_iterations: 1
morph_closing_iterations: 5

# Blob detection
min_circularity: 0.5
min_convexity: 0.25
min_inertia_ratio: 0.25

# Plant filter decision accumulators
min_accumulator_size: 10
max_accumulator_size: 100

